### PR TITLE
ODC-7422: Add SBOM link and signed badge in PLR details page

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 

https://issues.redhat.com/browse/ODC-7422

**Analysis / Root cause**: 

UI should surface SBOM information and command to download the sbom. Signed pipelineruns should show the badge next to pipelinerun details page.

**Solution Description**: 

Use taskrun's results to fetch the SBOM information and use the chain's annotation to show the signed badge.

**Screen shots / Gifs for design review**: 

image

---
**Tooltip on badge hover:**

Screenshot 2023-11-08 at 3 39 06 PM

**Test setup:**

1. Install openshift pipelines and create the resources listed in this [gist](https://gist.github.com/karthikjeeyar/4957a1cbb8437abe0d32f43b43a2b6c4)
2. Visit the pipelinerun detail page

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/PipelineRunDetailsPage.tsx [hunk 1]
```diff
@@ -23,14 +27,25 @@ const PipelineRunDetailsPage: React.FC<DetailsPageProps> = (props) => {
   );
   const breadcrumbsFor = useDevPipelinesBreadcrumbsFor(kindObj, match);
   const badge = usePipelineTechPreviewBadge(props.namespace);
-
+  const resourceTitleFunc = (obj: PipelineRunKind): string | JSX.Element =>
+    obj?.metadata?.annotations?.[chainsSignedAnnotation] === 'true' ? (
+      <div style={{ display: 'flex' }}>
+        {obj?.metadata?.name}{' '}
+        <Tooltip content={'Signed'}>
+          <img src={SignedPipelinerunIcon} alt="Signed" />
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelineruns/PipelineRunDetailsPage.tsx:35]: Please add i18n for Signed. :)

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx [hunk 2]
```diff
@@ -59,6 +65,33 @@ const PipelineRunCustomDetails: React.FC<PipelineRunCustomDetailsProps> = ({ pip
...
       </dl>
+      {buildImage && (
+        <dl>
+          <dt>{t('pipelines-plugin~Download SBOM')}</dt>
+          <dd>
+            <ClipboardCopy isReadOnly hoverTip="Copy" clickTip="Copied">
+              {`cosign download sbom ${buildImage}`}
+            </ClipboardCopy>
+            <ExternalLink href="https://docs.sigstore.dev/cosign/installation">
+              Install Cosign
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx:76]: Please add i18n for this as well

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx [hunk 3]
```diff
@@ -59,6 +65,33 @@ const PipelineRunCustomDetails: React.FC<PipelineRunCustomDetailsProps> = ({ pip
           <PipelineResourceRef {...convertBackingPipelineToPipelineResourceRefProps(pipelineRun)} />
         </dd>
       </dl>
+      {buildImage && (
+        <dl>
+          <dt>{t('pipelines-plugin~Download SBOM')}</dt>
+          <dd>
+            <ClipboardCopy isReadOnly hoverTip="Copy" clickTip="Copied">
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx:72]: i18n for both tooltips please ;)

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx [hunk 4]
```diff
@@ -59,6 +65,33 @@ const PipelineRunCustomDetails: React.FC<PipelineRunCustomDetailsProps> = ({ pip
...
+      {sbomTaskRun && (
+        <dl>
+          <dt>{t('pipelines-plugin~SBOM')}</dt>
+          <dd>
+            <Link
+              to={`/k8s/ns/${sbomTaskRun.metadata.namespace}/${referenceForModel(TaskRunModel)}/${
+                sbomTaskRun.metadata.name
+              }/logs`}
+            >
+              View SBOM
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx:90]: i18n

**Review by jerolimov — CHANGES_REQUESTED** [review]
Lgtm! Just some missed i18n strings:

File: frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts [hunk 5]
```diff
@@ -528,3 +528,21 @@ export const returnValidTaskModel = (task: TaskKind): K8sModel => {
...
+}
+
+export enum TaskRunResults {
+  SBOM = 'LINK_TO_SBOM',
+  SCAN_OUTPUT = 'SCAN_OUTPUT',
+  TEST_OUTPUT = 'TEST_OUTPUT',
+}
+
+export const getSbomTaskRun = (taskruns: TaskRunKind[]): TaskRunKind =>
+  taskruns?.find(
```
Comment (vdemeester) [inline frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts:543]: We don't read `task.results.format` anywhere do we ?

**Review by vdemeester — COMMENTED** [review]
Based on https://gist.github.com/karthikjeeyar/d9fe4d12fbbc1d01273c8540b5504d87#file-cve_scan_output_format-yaml-L7, should `task.results.format` be that generic (`application/json`) ? Because it's not like we are taking any `application/json` content, but that json content also need to "comply" to our "contract". 
I think that PR is only about the link and the "signed" badge, so that part has no effect, so it's more for another (upcoming or existing ?) PR.

File: frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts (hunk 5 above)
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts:543]: You are right @vdemeester , This PR just surfaces the SBOM link and reads annotations. The output tab PR will be using all other annotations like format, type etc

Comment (burrsutter) [reply frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts:543]: The CVE summary is also JSON based, is that part of this PR?

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx [hunk 6]
```diff
@@ -59,6 +65,37 @@ const PipelineRunCustomDetails: React.FC<PipelineRunCustomDetailsProps> = ({ pip
...
+              {`cosign download sbom ${buildImage}`}
+            </ClipboardCopy>
+            <ExternalLink href="https://docs.sigstore.dev/cosign/installation">
+              {t('pipelines-plugin~Install Cosign')}
+            </ExternalLink>
+          </dd>
+        </dl>
+      )}
+      {sbomTaskRun && (
+        <dl>
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx:86]: There is no need to start new lists (`dl`) for each definition (`dt+dd`).

Can you clean this up at least for your definitions, and have one list for the Pipeline definition, SBOM, View SBOM, and the following three definitions (Start time, Completion time and Duration)?

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx [hunk 7]
```diff
@@ -59,6 +65,37 @@ const PipelineRunCustomDetails: React.FC<PipelineRunCustomDetailsProps> = ({ pip
...
+          <dt>{t('pipelines-plugin~SBOM')}</dt>
+          <dd>
+            <Link
+              to={`/k8s/ns/${sbomTaskRun.metadata.namespace}/${referenceForModel(TaskRunModel)}/${
+                sbomTaskRun.metadata.name
+              }/logs`}
+            >
+              {t('pipelines-plugin~View SBOM')}
+            </Link>
```
Comment (jerolimov) [inline frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx:95]: I don't know your designs, but a `ResourceLink` might be a good option here. Wdyt?

**Review by jerolimov — CHANGES_REQUESTED** [review]
As said, it's almost good. Just one thing and one option/question:

File: frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx (hunk 7 above)
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/pipelineruns/detail-page-tabs/PipelineRunCustomDetails.tsx:95]: @jerolimov This is a taskruns logs link, I think ResourceLink is mainly used for redirecting to resource details page

**Comment by jerolimov** [comment]
@karthikjeeyar to get `jira/valid-reference` :smirk: 

/retitle ODC-7422: Add SBOM link and signed badge in PLR details page

**Review by jerolimov — APPROVED** [review]
Code lgtm, and it works as expected (as described).

@vdemeester @burrsutter when your comment is resolved we are happy to merge this.

/lgtm

**Comment by sanketpathak** [comment]
@karthikjeeyar Will you be adding any e12e tests for the change?

**Comment by karthikjeeyar** [comment]
@sanketpathak I have logged a story to cover e2e tests in separate PR

**Review by jerolimov — APPROVED** [review]

**Pull request merged**